src/Inciter/AMR/node_store.h

```diff
@@ -0,0 +1,245 @@
+#ifndef AMR_node_store_h
+#define AMR_node_store_h
+
+#include <cmath>
+
+#include "types.h"
+#include "tet_store.h"
+
+// TODO: make this have a base class to support multiple generator schemes
+// using the policy design pattern
+namespace AMR {
+
+    class node_store_t
+    {
+
+        private:
+            coord_type* m_x;
```

Comment:
See above. This was done like this a temporary measure to soft-wrap mesh_t, but to provide a (partial) abstraction from some of it's implementation details. It's definitely not the best solution ever!